pulsar-client/src/test/java/org/apache/pulsar/client/impl/schema/KeyValueSchemaTest.java

```diff
@@ -400,4 +404,18 @@ public void testKeyValueSchemaSeparatedEncoding() {
                 AutoConsumeSchema.getSchema(keyValueSchema.getSchemaInfo());
         assertEquals(keyValueSchema.getKeyValueEncodingType(), keyValueSchema2.getKeyValueEncodingType());
     }
+
+    @Test
+    public void testSchemaCaches() {
+        Schema<Foo> keySchema = spy(Schema.AVRO(Foo.class));
+        Schema<Foo> valueSchema = spy(Schema.AVRO(Foo.class));
+        KeyValueSchemaImpl<Foo, Foo> keyValueSchema = (KeyValueSchemaImpl<Foo,Foo>)
+                KeyValueSchemaImpl.of(keySchema, valueSchema, KeyValueEncodingType.SEPARATED);
+
+        keyValueSchema.atSchemaVersion(new byte[0]);
+        keyValueSchema.atSchemaVersion(new byte[0]);
```

Comment:
can we assert that we are returning the same object ?

Reply:
Sure